R/process_data.R

```diff
@@ -181,6 +186,73 @@ set_initial_plan_carsten_missings_to_zero <- function(data,
   return(data)
 }
 
+#' Remove rows from PACTA results that belong to company-sector combinations
+#' for which there is no positive production value in the relevant year of
+#' exposure (last year of forecast). This handles the edge case that a company
+#' may have a positive exposure for this sector, but none of the technologies
+#' covered in this analysis have any positive production. Such inconsistencies
+#' may arise e.g. because of unclear separation of the LDV and HDV sectors.
+#'
+#' @inheritParams calculate_annual_profits
+#' @inheritParams report_company_drops
+#' @param data tibble containing filtered PACTA results
+#'
+#' @return A tibble of data without rows with no exposure info
+#' @noRd
+remove_sectors_with_missing_production <- function(data,
+                                                   start_year,
+                                                   time_horizon,
+                                                   log_path) {
+  n_companies_pre <- length(unique(data$company_name))
+
+  # we merge the exposure of the last year of the forecast on the company
+  # results for the aggregation, to get the closest picture to the shock year.
+  # Hence start_year + time_horizon
+  companies_missing_sector_production <- data %>%
+    dplyr::filter(.data$year == .env$start_year + .env$time_horizon) %>%
+    dplyr::group_by(
+      .data$investor_name, .data$portfolio_name, .data$company_name,
+      .data$scenario, .data$scenario_geography, .data$ald_sector
+    ) %>%
+    dplyr::summarise(
+      sector_prod = sum(.data$plan_tech_prod, na.rm = TRUE),
+      .groups = "drop"
+    ) %>%
+    dplyr::ungroup() %>%
+    dplyr::filter(.data$sector_prod <= 0)
+
+  data_filtered <- data %>%
+    dplyr::anti_join(
+      companies_missing_sector_production,
```

Comment:
robustness:
in l 214 you group by 6 varibales, in l 227 you merge by only 2 variables. This look incorrect from the pattern. E.g. you would remove rows for investor A, even if only investor B has the problem, or if only 1 scenario_geography is affected.

I would suggest to check if all the grouping is really neccessary and removing it if possible, if not please include the variables in merging, because otherwise this calls for future bugs. I would argue however, that if the problem exists in 1 scenario also the row for the other scenario should be removed.

Reply:
okay, should be done now

Reply:
Just to go sure that it is really fine:

- did you verify that for all variables that you do not group by any longer there is some explicit filtering happning before?
- expectations: let us assume that companies_missing_sector_production holds a row for scenario A but not scenario B. is it correct behavior that in this case we would only remove the row for scenario A? I am nearly sure that it is not